checks/exec.go

```diff
@@ -49,14 +54,87 @@ func execPowershell(check v1.ExecCheck, ctx *context.Context) pkg.Results {
 	if err != nil {
 		result.Failf("powershell not found")
 	}
-	args := []string{*check.Script}
+	args := []string{check.Script}
 	cmd := osExec.Command(ps, args...)
 	return runCmd(cmd, result)
 }
 
 func execBash(check v1.ExecCheck, ctx *context.Context) pkg.Results {
 	result := pkg.Success(check, ctx.Canary)
-	cmd := osExec.Command("bash", "-c", *check.Script)
+	fields := strings.Fields(check.Script)
+	if len(fields) == 0 {
+		return []*pkg.CheckResult{result.Failf("no script provided")}
+	}
+
+	cmd := osExec.Command("bash", "-c", check.Script)
+
+	// Setup connection details
+	switch strings.ToLower(fields[0]) {
+	case "aws":
+		if check.Connections.AWS == nil {
+			return []*pkg.CheckResult{result.Failf("no AWS connection provided")}
+		}
+
+		if err := check.Connections.AWS.Populate(ctx, ctx.Kubernetes, ctx.Namespace); err != nil {
+			return []*pkg.CheckResult{result.Failf("failed to hydrate aws connection: %v", err)}
+		}
+
+		configPath, err := saveConfig("aws-*", awsConfigTemplate, check.Connections.AWS)
+		defer os.RemoveAll(filepath.Dir(configPath))
+		if err != nil {
+			return []*pkg.CheckResult{result.Failf("failed to store AWS credentials: %v", err.Error())}
+		}
+
+		cmd.Env = os.Environ()
+		cmd.Env = append(cmd.Env, "AWS_EC2_METADATA_DISABLED=true") // https://github.com/aws/aws-cli/issues/5262#issuecomment-705832151
+		cmd.Env = append(cmd.Env, fmt.Sprintf("AWS_SHARED_CREDENTIALS_FILE=%s", configPath))
+		if check.Connections.AWS.Region != "" {
+			cmd.Env = append(cmd.Env, fmt.Sprintf("AWS_DEFAULT_REGION=%s", check.Connections.AWS.Region))
+		}
+
+	case "az":
```

Comment:
I think use `azure`

Reply:
I think `az` would be the standard way to use the azure cli. I have azure cli installed locally, and I can only access it with `az`. Unless we setup `alias azure='az'`

Reply:
`az` maybe the command name, but it isn't clear it means azure.